### PR TITLE
Restore running Maven checks by default

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

Restore running Maven checks by default.

Since it was disabled for two days, also fix:
* dependency issues after adding `trino-exchange` to `trino-tests`;
* imports order.

> Is this change a fix, improvement, new feature, refactoring, or other?
other

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)
tests

> How would you describe this change to a non-technical end user or system administrator?
n/a

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: client/trino-jdbc/pom.xml [hunk 1]
```diff
@@ -121,6 +121,14 @@
             <groupId>io.trino</groupId>
             <artifactId>trino-testing</artifactId>
             <scope>test</scope>
+
+            <exclusions>
```
Comment (losipiuk) [inline client/trino-jdbc/pom.xml:125]: nit: commit message to long

**Review by losipiuk — APPROVED** [review]
LGTM.
Yet as discussed offline it may be better to add `trino-exchange` as a provided dependency of `trino-testing`. And add it for real only to modules which actually make use of the classes from there (trino-tests and maybe sth else).

I did not verify if this approach would work - but seems as it should.

@hashhar , @findepi  opinions?

File: plugin/trino-thrift-testing-server/pom.xml [hunk 2]
```diff
@@ -32,6 +32,14 @@
         <dependency>
             <groupId>io.trino</groupId>
             <artifactId>trino-testing</artifactId>
+
+            <exclusions>
+                <exclusion>
+                    <!-- not needed for thrift-testing-server, and results with conflict regarding netty -->
+                    <groupId>io.trino</groupId>
+                    <artifactId>trino-exchange</artifactId>
```
Comment (findepi) [inline plugin/trino-thrift-testing-server/pom.xml:40]: does trino-exchange use netty?

File: plugin/trino-hive-hadoop2/pom.xml [hunk 3]
```diff
@@ -140,6 +140,17 @@
...
                         </configuration>
                     </plugin>
+                    <plugin>
+                        <groupId>org.basepom.maven</groupId>
+                        <artifactId>duplicate-finder-maven-plugin</artifactId>
+                        <configuration>
+                            <ignoredResourcePatterns>
+                                <!-- com.amazonaws:aws-java-sdk-s3 and software.amazon.awssdk:sdk-core MIME type file duplicate-->
+                                <ignoredResourcePattern>mime.types</ignoredResourcePattern>
+                                <ignoredResourcePattern>about.html</ignoredResourcePattern>
```
Comment (findepi) [inline plugin/trino-hive-hadoop2/pom.xml:150]: And these are not identical, right?

**Comment by findepi** [comment]
> Yet as discussed offline it may be better to add `trino-exchange` as a provided dependency of `trino-testing`. And add it for real only to modules which actually make use of the classes from there (trino-tests and maybe sth else).

I would suggest removing it from there.
I.e redo (or revert) a0d8caa519f14aee44da90a552ce27f88dbe69e3 cc @losipiuk

**Comment by findepi** [comment]
Let me try: https://github.com/trinodb/trino/pull/11311

File: plugin/trino-hive-hadoop2/pom.xml (hunk 3 above)
Comment (nineinchnick) [reply plugin/trino-hive-hadoop2/pom.xml]: Removed since I rebased over #11311

**Review by losipiuk — APPROVED** [review]

**Comment by findepi** [comment]
> Can you please rebase on top of #11311 ?

Can you please do this again? #11311 was broken.

File: plugin/trino-cassandra/pom.xml [hunk 4]
```diff
@@ -157,13 +157,6 @@
             <groupId>io.trino</groupId>
             <artifactId>trino-testing</artifactId>
             <scope>test</scope>
-            <exclusions>
-                <exclusion>
-                    <!-- not needed for pinot, and results with conflict regarding netty and apache httpclient -->
-                    <groupId>io.trino</groupId>
-                    <artifactId>trino-exchange</artifactId>
```
Comment (findepi) [inline plugin/trino-cassandra/pom.xml:164]: This belongs to "Remove trino-testing dependency on trino-exchange". Can you please squash?

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
@nineinchnick see red CI

File: plugin/trino-cassandra/pom.xml (hunk 4 above)
Comment (nineinchnick) [reply plugin/trino-cassandra/pom.xml:164]: Done!

**Comment by nineinchnick** [comment]
> @nineinchnick see red CI

I swear I ran a build locally to check it first ... Rebased on the latest master and I checked locally again.

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
added fixes for new imports ordering violations from #11312

**Pull request merged**